### ISSUE TITLE
2020-05-14 tc1513 M2SD weight-normalised parameters provided when NORMBSU is not valid

### ISSUE THREAD

**Issue by jhhughes256** [tensfeldt/openNCA#247]
### Issue

When running tc1513 M2SD (and most other testcases) with mct$NORMBSU set to an invalid column name, weight-normalised parameters are still provided in CE output.

When no valid NORMBSU value is given, weight-normalised parameters should be NA.

### Description

To replicate this, the following changes can be made to the mct.

```r
mct$NORMBSU <- NULL
# or
mct$NORMBSU <- "KG"
```

When running with an invalid NORMBSU value a warning is given which is good.

```r
> results_list <- run_computation(data = d, map = mct, flag = flags, 
+   parameterset = parameterset)
unit_conversion : Unit Class 1 (Time) time_col:  TMAX TMIN TLAST KELTMLO KELTMHI THALF LASTTIME MRTLAST MRTIVIFO MRTIVIFP  parameters are scaled from  HR  to  HR  via scaling factor:  1 
unit_conversion : Unit Class 3 (Dose) dose_col:  DOSE1  parameters are scaled from  UG  to  MG  via scaling factor:  0.001 
unit_conversion : Unit Class 4 (Volume) volume_col:  V0 VZO VZP VSSO VSSP  parameters are scaled from  ML  to  L  via scaling factor:  0.001 
unit_conversion : Unit Class 5 (Amount/Volume) conc_col:  C0 CMAX CMIN CLAST CMAXC CEST KELC0  parameters are scaled from  NG/ML  to  NG/ML  via scaling factor:  1 
unit_conversion : Unit Class 6 (1/Time) kel_col:  KEL  parameters are scaled from  1/HR  to  1/HR  via scaling factor:  1 
unit_conversion : Unit Class 7 (Volume/Time) cl_col:  CLO CLP  parameters are scaled from  ML/HR  to  ML/HR  via scaling factor:  1 
unit_conversion : Unit Class 8: (Amount.Time/Volume) auc_col:  AUCALL AUCLAST AUCLASTC AUC1 AUC2 AUC3 AUC4 AUC5 AUC6 AUC7 AUC8 AUC9 AUC10 AUC11 AUC12 AUC13 AUC14 AUC15 AUC16 AUC17 AUC0_12 AUC24_504 AUC0_48 AUC0_20 AUC0_696 AUCINFO AUCINFP AUCINFOC AUCINFPC  parameters are scaled from  NG.HR/ML  to  NG.HR/ML  via scaling factor:  1 
unit_conversion : Unit Class 9: (Amount.Time.Time/Volume) aumc_col:  AUMCLAST AUMCINFO AUMCINFP  parameters are scaled from  NG.HR.HR/ML  to  NG.HR.HR/ML  via scaling factor:  1 
unit_conversion : Unit Class 10: ([Amount.Time/Volume]/Amount) aucdn_col:  AUCALLDN AUCLASTDN AUC1DN AUC2DN AUC3DN AUC4DN AUC5DN AUC6DN AUC7DN AUC8DN AUC9DN AUC10DN AUC11DN AUC12DN AUC13DN AUC14DN AUC15DN AUC16DN AUC17DN AUCINFODN AUCINFPDN  parameters are scaled from  NG.HR/ML/UG  to  NG.HR/ML/MG  via scaling factor:  1000 
unit_conversion : Unit Class 12: ([Amount/Volume]/Amount) concdn_col:  CMAXDN  parameters are scaled from  NG/ML/UG  to  NG/ML/MG  via scaling factor:  1000 
Warning messages:
1: In validate_timeconc_data(map, data, flag, verbose = verbose) :
  No tau information provided in 'map'.
2: In validate_timeconc_data(map, data, flag, verbose = verbose) :
  No told information provided in 'map'.
3: In run_computation(data = d, map = mct, flag = flags, parameterset = parameterset) :
  'TAUs and TOLDs values are not provided via 'flag'! Using values provided via 'data' instead (if present)!
4: In run_M2_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTTAU' cannot be computed if 'TAUi' is not provided
5: In run_M2_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGEMESIS' is not present in the dataset
6: In unit_conversion(data = data_data, map = map_data, result = computation_df,  :
  'AMOUNTU' #2# is not present in the dataset provided via 'map'
7: In unit_conversion(data = data_data, map = map_data, result = computation_df,  :
  'PKPTMU' and 'CONCU' #10# are not present in the dataset provided via 'map'
8: In unit_conversion(data = data_data, map = map_data, result = computation_df,  :
  'CONCU' and 'NORMBSU' #14# are not present in the dataset provided via 'map'
9: In unit_conversion(data = data_data, map = map_data, result = computation_df,  :
  'CONCU', 'PKPTMU and 'NORMBSU' #19# are not present in the dataset provided via 'map'
```

But the weight-normalised parameters are still provided in the CE output despite unit conversion not occurring and no value being present in the VOLUMEWU and CLWU columns.

```r
> r[,c(grep("^CL.+(W|U)$", names(r)), grep("^V.+(W|WU)$", names(r)))]
         CLOW       CLPW CLWU      VZOW      VZPW     VSSOW     VSSPW VOLUMEWU
1   1.4896591  1.4908969   NA 127.93215 128.03846  94.10743  93.94937       NA
2   1.0254702  1.0292378   NA  68.28857  68.53947  75.75480  75.42537       NA
3   1.2874996  1.2890519   NA  99.86512  99.98552  76.75269  76.55623       NA
4   1.0957589  1.0946438   NA  70.30355  70.23201  61.74615  61.82569       NA
5   1.0867888  1.0954747   NA  91.75072  92.48403  76.53189  75.55004       NA
6   2.7326051  2.7300945   NA  76.33733  76.26719  76.96651  77.01562       NA
7   2.2161568  2.2200520   NA 125.70596 125.92690  97.34909  97.00202       NA
8   1.5218201  1.5218836   NA 136.55279 136.55848  90.48286  90.47404       NA
9   0.8597127  0.8594096   NA  92.88937  92.85662  72.17925  72.19746       NA
10  1.4150924  1.4279847   NA  53.56137  54.04935  53.08368  52.62620       NA
11  1.1848945  1.2136749   NA  50.90188  52.13826  53.43331  51.96412       NA
12  1.1577788  1.2433287   NA  53.22501  57.15788  62.29953  58.80336       NA
13  0.9694509  1.0533438   NA  63.63469  69.14141  79.54327  72.60882       NA
14  1.0675942  1.0683168   NA  88.82468  88.88480  71.87255  71.78823       NA
15  1.5219818  1.5693361   NA  57.35832  59.14294  62.02724  59.41946       NA
16  1.8612268  1.8616659   NA 105.32064 105.34549  88.88471  88.84910       NA
17  1.0370253  1.0374086   NA  98.63811  98.67457  80.83337  80.79225       NA
18  2.1236028  2.1465780   NA  53.99245  54.57659  51.50191  49.80239       NA
19  1.0519883  1.0624123   NA  51.51316  52.02359  51.58096  50.82948       NA
20 19.3843211 19.9078181   NA 264.47987 271.62247 267.82893 270.85378       NA
```

**Comment by TomTensfeldt** [comment]
Issue occurs with tc143_M1SS as well.

**Comment by jhhughes256** [comment]
This issue appears to have been addressed with commit cca452d. Closing this issue for now. tc1513 output below.

```r
>  r[,c(grep("^CL.+(W|U)$", names(r)), grep("^V.+(W|WU)$", names(r)))]
   CLOW CLPW CLWU VZOW VZPW VSSOW VSSPW VOLUMEWU
1    NA   NA   NA   NA   NA    NA    NA       NA
2    NA   NA   NA   NA   NA    NA    NA       NA
3    NA   NA   NA   NA   NA    NA    NA       NA
4    NA   NA   NA   NA   NA    NA    NA       NA
5    NA   NA   NA   NA   NA    NA    NA       NA
6    NA   NA   NA   NA   NA    NA    NA       NA
7    NA   NA   NA   NA   NA    NA    NA       NA
8    NA   NA   NA   NA   NA    NA    NA       NA
9    NA   NA   NA   NA   NA    NA    NA       NA
10   NA   NA   NA   NA   NA    NA    NA       NA
11   NA   NA   NA   NA   NA    NA    NA       NA
12   NA   NA   NA   NA   NA    NA    NA       NA
13   NA   NA   NA   NA   NA    NA    NA       NA
14   NA   NA   NA   NA   NA    NA    NA       NA
15   NA   NA   NA   NA   NA    NA    NA       NA
16   NA   NA   NA   NA   NA    NA    NA       NA
17   NA   NA   NA   NA   NA    NA    NA       NA
18   NA   NA   NA   NA   NA    NA    NA       NA
19   NA   NA   NA   NA   NA    NA    NA       NA
20   NA   NA   NA   NA   NA    NA    NA       NA
```
